web/profiles/joinup/src/Plugin/Block/AddContentBlock.php

```diff
@@ -33,7 +80,66 @@ public function build() {
       ],
     ];
 
+    // This check has to occur here so that the link can be cached correctly
+    // for each page.
+    if (
+      !($this->account->isAnonymous())
```

Comment:
The link access is handled by `CustomPageController::access()` so you don't need to check it here.